test/controllers/account/passwords_controller_test.rb

```diff
@@ -1,52 +1,49 @@
 require "test_helper"
 
 class Account::PasswordsControllerTest < ActionController::TestCase
-  def change_user_password(user_factory, new_password)
-    original_password = "I am a very original password. Refrigerator weevil."
-    user = create(user_factory, password: original_password)
-    original_password_hash = user.encrypted_password
-    sign_in user
-
-    post :update,
-         params: {
-           user: {
-             current_password: original_password,
-             password: new_password,
-             password_confirmation: new_password,
-           },
-         }
-
-    [user, original_password_hash]
+  setup do
+    @original_password = SecureRandom.hex
+    @user = create(:user, password: @original_password)
+    sign_in @user
   end
 
   context "PUT update" do
-    should "changing passwords to something strong should succeed" do
-      user, orig_password = change_user_password(:user, "destabilizers842}orthophosphate")
+    should "update password if password is sufficiently strong" do
+      new_password = SecureRandom.hex
```

Comment:
Good point - I think the server-side Devise check *is* triggered in controller tests, so your concern is valid.

In another test somewhere, I've used the method that `devise_invitable` uses internally for setting a default password, i.e. [`User.send(:random_password)`](https://github.com/scambra/devise_invitable/blob/db2f9a1c0e87addb196d1b82920370f4230c1157/lib/devise_invitable/models.rb#L395-L405). The comment for that it covers most formatting rules, but since `devise_zxcvbn` is a separate gem, maybe it won't always cope with that validation.

I see that you've recently used `SecureRandom.urlsafe_base64` elsewhere. Do you think that doesn't suffer from the same problem?

Perhaps the simplest thing is to pick a constant strong password rather than trying to generate one every time...?

Reply:
I've resolved this by hardcoding the passwords in the tests to random values that pass the validation. I've amended the commits and force-pushed.